app/code/Vaimo/QtyManager/view/frontend/web/js/qty-manager.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright © Vaimo, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+'jquery',
+'underscore',
+'mage/template',
+'text!Vaimo_QtyManager/template/qty-template.html',
+'jquery/ui',
+'mage/validation'
+   ], function ($, _, template, qtyTemplate) {
+    'use strict';
+
+    $.widget('mage.qtyManager', {
+        _create: function () {
+            this._constructorBlock();
+            this._minQtyChecker();
+            this._bind();
+        },
+
+        /**
+         * @description Method for adding increase/decrease buttons template and variable declaration.
+         * And replace quantity input to this template.
+         */
+        _constructorBlock: function () {
+            this.element.after($(template(qtyTemplate, '')));
+            this.block = this.element.next('[data-qty-wrapper]');
+            this.block.find('[data-input]').replaceWith(this.element);
+            this.qtyInput = this.block.find('.input-text.qty');
+            this.decreaseBtn = this.block.find('[data-qty=decrease]');
+        },
+
+        /**
+         * @description Method for getting current qty value.
+         * @return {number} - quantity from input
+         */
+        _getQnty: function () {
+            return +this.qtyInput.val();
+        },
+
+        /**
+         * @description Method for setting new qty value.
+         * @param {qty} qty - The new quantity value.
+         */
+        _setQty: function (qty) {
+            this.qtyInput.val(qty);
+            this.triggerChangeInput();
+        },
+
+        /**
+         * @description Method for update qty value and total price.
+         * @param {qty} qty - The new quantity value.
+         */
+        _updateQty: function (qty) {
+            this._setQty(qty);
+            // Apply triggerChangeInput on minicart
+            if (this.qtyInput.closest('#mini-cart').length) {
+                this.triggerChangeInput();
+            }
+        },
+
+        /**
+         * @description Method for decrease and increase events.
+         */
+        _bind: function () {
+            let self = this;
+            this.block.on('click', '[data-qty=decrease]', function (e) {
```

Comment:
[data-qty=decrease], can be moved in widget options and used everywhere in widget from options.